boskos/client/client.go

```diff
@@ -93,6 +94,28 @@ func (c *Client) Acquire(rtype, state, dest string) (*common.Resource, error) {
 	return r, nil
 }
 
+// AcquireWait blocks until Acquire returns the specified resource or the
+// provided context is cancelled or its deadline exceeded.
+func (c *Client) AcquireWait(ctx context.Context, rtype, state, dest string) (*common.Resource, error) {
+	// Try to acquire the resource until available or the context is
+	// cancelled or its deadline exceeded.
+	for {
+		r, err := c.Acquire(rtype, state, dest)
+		if err != nil {
+			if err == ErrAlreadyInUse || err == ErrNotFound {
+				if ctx != nil && ctx.Err() == nil {
+					time.Sleep(time.Second * 3)
+					if ctx.Err() == nil {
```

Comment:
Hi @sebastienvas,

Do you think the second check is necessary? I put it here, but I'm having second doubts as the more error could also be the context error at this point. Thoughts?